WalletWasabi.Documentation/Guides/DeterministicBuildGuide.md

```diff
@@ -1,67 +1,79 @@
+
+## 1. Assert correct environment
+
+In order to reproduce Wasabi's builds, you need [git](https://git-scm.com/) package, Windows 10, and the version of .NET Core SDK that was used by the Wasabi team to produce given Wasabi Wallet release. The latest version of .NET Core SDK is always used, unless specified otherwise in the release notes of Wasabi Wallet. You can get it here https://dotnet.microsoft.com/download/dotnet-core.
+
+## 2. Reproduce builds
+
+You can see the list of Wasabi releases here: https://github.com/zkSNACKs/WalletWasabi/releases. Please note that each release has a version and a git hash assigned. The git hash is useful in the following instructions:
 
 ```sh
 git clone https://github.com/zkSNACKs/WalletWasabi.git
-git checkout {hash of the release} # This works from 1.1.3 release, https://github.com/zkSNACKs/WalletWasabi/releases
-cd WalletWasabi/WalletWasabi.Packager/
+cd WalletWasabi
+git checkout <SHA-1-hash-of-the-release>
+cd WalletWasabi.Packager
 dotnet restore
 dotnet build
 dotnet run -- --onlybinaries
 ```
 
-This will build our binaries for Windows, OSX and Linux from source code and open them in a file explorer for you.
+These commands will produce Wasabi's binaries for Windows, macOS and Linux from source code. Also, for your convenience, a new file explorer window will navigate you to the binaries location - i.e. `WalletWasabi\\WalletWasabi.Gui\\bin\\dist`.
 
 ![](https://i.imgur.com/8XAQzz4.png)
 
-# 3. Verify Builds
+## 3. Verify builds
 
-You can compare our binaries with the downloads we have on the website: https://wasabiwallet.io/
-In order to end-to-end verify all the downloaded packages you need a Windows, a Linux, and an OSX machine.
+Now, we will attempt to verify the binaries you have just compiled with the officially distributed binaries on https://wasabiwallet.io website. Please download those packages from the website, you should see the following files in your File Explorer:
 
 ![](https://i.imgur.com/aI9Kx0c.png)
 
-## Windows
+Please note that to completely verify Wasabi packages for all supported platforms, you actually need machines with Windows, Linux and macOS operating systems. If you don't have so many physical machines, you can use any virtualization sofware packages like [VirtualBox](https://www.virtualbox.org/), [VMWare](https://www.vmware.com/), etc.
 
-After you installed Wasabi from the `.msi`, it will be in `C:\Program Files\WasabiWallet` folder. You can compare it with your build:
+### Windows
 
-```sh
-git diff --no-index win7-x64 "C:\Program Files\WasabiWallet"
-```
+* Install Wasabi using `Wasabi-<version>.msi` file. It will install to `C:\Program Files\WasabiWallet` directory.
+* Start `cmd` or Powershell and navigate to the `dist` directory.
+* Execute the following command:
+  ```sh
+  git diff --no-index "win7-x64" "C:\Program Files\WasabiWallet"
```

Comment:
I am not familiar enough with the procedure but if the user needs wsl anyway, why Powershel and why `git diff` instead of simply `diff`?

Reply:
WSL is optional feature of Windows 10 that needs to be installed, reboot is required. You may not want to verify builds for all platforms if you are an ordinary Wasabi user. If you do have WSL installed, you can verify Windows build there too, that's correct.

Powershell is getting traction on Windows platform, so I don't see a reason why not to use it, given how powershell that thing is.

`git diff` is used probably because you don't want to install any additional commands on Windows platform. It behaves, AFAIK, the same on all platforms so using simple `diff` would complicate things further.

Reply:
Ok, I got it.